server/datastore/mysql/migrations/tables/20250502222222_AddMdmEnrollTables.go

```diff
@@ -0,0 +1,259 @@
+package tables
+
+import (
+	"database/sql"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/fleetdm/fleet/v4/server/datastore/mysql/common_mysql"
+	"github.com/fleetdm/fleet/v4/server/fleet"
+	"github.com/jmoiron/sqlx"
+	"github.com/jmoiron/sqlx/reflectx"
+)
+
+func init() {
+	MigrationClient.AddMigration(Up_20250502222222, Down_20250502222222)
+}
+
+func Up_20250502222222(tx *sql.Tx) error {
+	txx := sqlx.Tx{Tx: tx, Mapper: reflectx.NewMapperFunc("db", sqlx.NameMapper)}
+
+	// legacy_host_mdm_enroll_refs captures existing enroll refs from the host_mdm table. Going
+	// forward, it is used to ensure the enroll refs is appended to the server URL when legacy Apple
+	// devices attempt to re-enroll. This table eventually could be dropped once all legacy devices
+	// are retired.
+	createStmt := `
+CREATE TABLE IF NOT EXISTS legacy_host_mdm_enroll_refs (
+	id INT UNSIGNED NOT NULL AUTO_INCREMENT PRIMARY KEY,
+	host_uuid VARCHAR(36) NOT NULL,
+	enroll_ref VARCHAR(36) NOT NULL,
+	INDEX idx_legacy_enroll_refs_host_uuid (host_uuid)
```

Comment:
Should that be unique per host? Or could there be multiple and they are all valid?

Reply:
There aren't great unique constraints in place currently so it is possible for there to be more than one. My thinking is to go ahead and preserve whatever is there for now so we'll have it available if we encounter issues with any legacy deployments.

Under the new approach, we will enforce one-per-host uuid uniqueness in the `host_mdm_idp_accounts` table so hopefully that will be easier to manage going forward.